tools/lsp/preview/ui/property_view.rs

```diff
@@ -178,19 +180,166 @@ fn convert_number_literal(
+            return extract_color_from_evaluated_expression_impl(from, palette_name, struct_level);
+        }
+        expression_tree::Expression::NumberLiteral(value, unit)
+            if *unit == expression_tree::Unit::None =>
+        {
+            println!("Interpreting NumberLiteral as Color");
+            return ExtractedColor {
+                value: Some(slint::Color::from_argb_encoded(*value as u32)),
+                name: if palette_name.is_empty() {
+                    None
+                } else {
+                    Some(palette_name.iter().rev().join("."))
+                },
+            };
+        }
+        expression_tree::Expression::PropertyReference(source)
+            if source.element().borrow().base_type == langtype::ElementType::Global =>
+        {
+            println!("Following property reference");
+            let elem = source.element();
+            let elem = elem.borrow();
+            if let Some(component_name) = elem.enclosing_component.upgrade().map(|c| c.id.clone()) {
+                if let Some(binding) = elem.bindings.get(source.name()) {
+                    palette_name.push(source.name().to_string());
+                    palette_name.push(component_name.to_string());
+
+                    let struct_level = palette_name.len().saturating_sub(3);
+
+                    return extract_color_from_evaluated_expression_impl(
+                        &binding.borrow().expression,
+                        palette_name,
+                        struct_level,
+                    );
+                }
+            }
+        }
+        expression_tree::Expression::StructFieldAccess { base, name } => {
+            println!("Stripping StructFieldAccess");
+            palette_name.push(name.to_string());
+            return extract_color_from_evaluated_expression_impl(base, palette_name, struct_level);
+        }
+        expression_tree::Expression::Struct { values, .. } => {
+            println!("Stripping Struct");
+            if let Some(part) = palette_name.get(struct_level) {
+                eprintln!("  => field: {part}");
+                if let Some(value) = values.get(&SmolStr::from(part)) {
+                    eprintln!("    => with expr: {value:?}");
+                    return extract_color_from_evaluated_expression_impl(
+                        value,
+                        palette_name,
+                        struct_level.saturating_sub(1),
+                    );
+                }
+            }
+        }
+        expression_tree::Expression::Condition { true_expr, .. } => {
+            println!("Stripping Condition");
+            return extract_color_from_evaluated_expression_impl(
+                true_expr,
+                palette_name,
+                struct_level,
+            );
+        }
+        expression_tree::Expression::CodeBlock(cb) => {
+            println!("Stripping CodeBlock");
+            let Some(expr) = cb.first() else {
```

Comment:
```suggestion
            let Some(expr) = cb.last() else {
```

In case this is a `{ debug(foo); return blue; }`  for example,  the last value is the value that counts.

Reply:
I need to test whether that still works for the case I actually need to support.

Reply:
This is the Codeblock I run into:

```
CodeBlock(
    [
        StoreLocalVariable {
            name: "tmpobj_conv_0",
            value: Condition {
                condition: BinaryExpression {
[...]
                },
                true_expr: Struct {
[...]
                },
                false_expr: Struct {
[...]
        },
        Struct {
[...]
        },
    ],
)
```

for the code in line 1328 of `property_view.rs`:

```slint
    out property <Baz> s: foo == 0 ? { baz: { bar: Colors.blue } } : { baz: { bar: Colors.blue } };
```

I do want the Condition, which is the first element.

Reply:
That's strange. I'll debug that.

Reply:
This code supports a small subset of all possible expressions. I think we will be fine without reimplementing more of the evaluation logic for now.

The worst that can happen is that some palette entry is "too complex" and we will skip it. I am fine with that as long as that does not happen with the code our Figma plugin creates.

I admit I have not tested the code generated from Figma extensively yet (I just looked at the generated files and extracted things that caught my attention into test cases), so I will probably need to extend this before I am done.

Reply:
> The worst that can happen is that some palette entry is "too complex" and we will skip it

Or that this code things there is a color and just gives some random intermediate value instead.